bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -291,6 +381,106 @@ where
 			[GlobalConsensus(network), AccountKey20 { network: None, key: token.into() }],
 		)
 	}
+
+	/// Constructs an XCM message destined for AssetHub that withdraws assets from the sovereign
+	/// account of the Gateway contract and either deposits those assets into a recipient account or
+	/// forwards the assets to another parachain.
+	fn convert_send_native_token(
+		message_id: H256,
+		chain_id: u64,
+		token_id: TokenId,
+		destination: Destination,
+		amount: u128,
+		asset_hub_fee: u128,
+	) -> Result<(Xcm<()>, Balance), ConvertMessageError> {
+		let network = Ethereum { chain_id };
+		let asset_hub_fee_asset: Asset = (Location::parent(), asset_hub_fee).into();
+
+		let (dest_para_id, beneficiary, dest_para_fee) = match destination {
+			// Final destination is a 32-byte account on AssetHub
+			Destination::AccountId32 { id } =>
+				(None, Location::new(0, [AccountId32 { network: None, id }]), 0),
+			// Final destination is a 32-byte account on a sibling of AssetHub
+			Destination::ForeignAccountId32 { para_id, id, fee } =>
+				(Some(para_id), Location::new(0, [AccountId32 { network: None, id }]), fee),
+			// Final destination is a 20-byte account on a sibling of AssetHub
+			Destination::ForeignAccountId20 { para_id, id, fee } =>
+				(Some(para_id), Location::new(0, [AccountKey20 { network: None, key: id }]), fee),
+		};
+
+		let total_fees = asset_hub_fee.saturating_add(dest_para_fee);
+		let total_fee_asset: Asset = (Location::parent(), total_fees).into();
+
+		let asset_loc =
+			ConvertAssetId::convert(&token_id).ok_or(ConvertMessageError::InvalidToken)?;
+
+		let mut reanchored_asset_loc = asset_loc.clone();
+		reanchored_asset_loc
+			.reanchor(&GlobalAssetHubLocation::get(), &UniversalLocation::get())
+			.map_err(|_| ConvertMessageError::CannotReanchor)?;
```

Comment:
AFAICT this is incorrect:

`asset_loc` is location relative to Ethereum (that's what the pallet stores in `token_id <> location_relative_to_ethereum` mapping)

and you are trying to reanchor to AH as if it is relative to `UniversalLocation::get()` which is BH location here, not Ethereum.

if this is passing tests by accident, make sure to add regression test too when fixing it

Reply:
Nice catch! We do have tests before which cover different cases, but seems context in BH output the same result as context in Ethereum.

https://github.com/paritytech/polkadot-sdk/pull/5546/commits/c94da518d0165a222d978925784ddea3b0c2fa19 for the fix.

Reply:
Agree, option 2 make more sense here.

But we do need to store the location relative to Ethereum as in `outbound-router` [here](https://github.com/Snowfork/polkadot-sdk/blob/de6d3ee61759711c10f4a88d4c8eaffe78edf1d7/bridges/snowbridge/primitives/router/src/outbound/mod.rs#L395-L399) we need to check the location of the asset which is reanchored on AH in context of Ethereum [here](https://github.com/paritytech/polkadot-sdk/blob/2d4e89f091c103813e462bd9cb188473b2673708/polkadot/xcm/pallet-xcm/src/lib.rs#L1865-L1869) already registed.

TBH I'm not sure the reanchor logic on AH even make sense. For our use case the bridge transfer actually terminates on BH, we won't send the xcm to Ethereum. So the destination should be BH IMO.

Anyway, to limit the scope I prefer option 1 as a workaround and we can improve it later.
https://github.com/paritytech/polkadot-sdk/pull/5546/commits/116c421d1d56dc43923d6af01b5d5f7b38678099